src/app/store/effects/product.effects.ts

```diff
@@ -79,6 +80,15 @@ export class ProductsEffects {
     })
   );
 
+  @Effect()
+  deleteProductFail$ = this.actions$.pipe(
+    ofType<DeleteProductFail>(EProductActions.DeleteProductFail),
+    switchMap(() => {
+      const message = 'Something went wrong while deleting of the product';
+      return of(new OpenModalAction(message));
+    }),
+  );
+
```

Comment:
The `createProductFail$` and the `deleteProductFail$` effects is very similar. We need to create common effect for processing the error cases. It will be good point.

Reply:
All good. Fixed